### DISCUSSION TITLE
The concept of "user..."

### DISCUSSION THREAD

**Discussion by mcctuxic** [tabemann/zeptoforth#124] [Q&A]
I have a question... :)

In the wordlist i found for example this two:
cuser and friends
user
2user, user-buffer:, user-aligned-buffer

in opposite to
cvariable
hvariable
2variable, buffer:, aligned-buffer

and more.

What is the difference between these two groups, the "user"-ones and the "non-user"-ones.

I searched the word lists in the docs and searched for "user" in the Getting Started docu without success.

**Comment by tabemann** [comment]
On Wed, Jul 31, 2024 at 11:56 AM mcctuxic ***@***.***> wrote:

> I have a question... :)
>
> In the wordlist i found for example this two:
> cuser and friends
> user
> 2user, user-buffer:, user-aligned-buffer
>
> in opposite to
> cvariable
> hvariable
> 2variable, buffer:, aligned-buffer
>
> and more.
>
> What is the difference between these two groups, the "user"-ones and the
> "non-user"-ones.
>
> I searched the word lists in the docs and searched for "user" in the
> Getting Started docu without success.
>
The USER words compile to flash variables or buffers that are task-local
and exist separately for each task. Note that after creating them one ought
to reboot before using said variables or buffers or creating more tasks.

Travis

**Comment by mcctuxic** [comment]
On 07/31 10:54, tabemann wrote:
> On Wed, Jul 31, 2024 at 11:56 AM mcctuxic ***@***.***> wrote:
> 
> > I have a question... :)
> >
> > In the wordlist i found for example this two:
> > cuser and friends
> > user
> > 2user, user-buffer:, user-aligned-buffer
> >
> > in opposite to
> > cvariable
> > hvariable
> > 2variable, buffer:, aligned-buffer
> >
> > and more.
> >
> > What is the difference between these two groups, the "user"-ones and the
> > "non-user"-ones.
> >
> > I searched the word lists in the docs and searched for "user" in the
> > Getting Started docu without success.
> >
> The USER words compile to flash variables or buffers that are task-local
> and exist separately for each task. Note that after creating them one ought
> to reboot before using said variables or buffers or creating more tasks.
> 
> Travis

Hi Travis,

thanks for the info! :)

...and I thought of "user" like the user in "user permissions"...I am a
Linux guy...sorry :)
